src/backend/distributed/commands/extension.c

```diff
@@ -115,3 +110,516 @@ ExtractNewExtensionVersion(Node *parsetree)
+ * clause.
+ * Cannot return NULL, errors out as Postgres would do if NULL.
+ */
+static char *
+GetCurrentSchema(void)
+{
+	/*
+	 * Neither user nor author of the extension specified schema; use the
+	 * current default creation namespace, which is the first explicit
+	 * entry in the search_path.
+	 */
+	List *search_path = fetch_search_path(false);
+	Oid schemaOid = InvalidOid;
+	char *schemaName = NULL;
+
+	if (search_path == NIL) /* nothing valid in search_path? */
+	{
+		ereport(ERROR,
+				(errcode(ERRCODE_UNDEFINED_SCHEMA),
+				 errmsg("no schema has been selected to create in")));
+	}
+	schemaOid = linitial_oid(search_path);
+	schemaName = get_namespace_name(schemaOid);
+	if (schemaName == NULL) /* recently-deleted namespace? */
+	{
+		ereport(ERROR,
+				(errcode(ERRCODE_UNDEFINED_SCHEMA),
+				 errmsg("no schema has been selected to create in")));
+	}
+
+	list_free(search_path);
+
+	return pstrdup(schemaName);
+}
+
+
+/*
+ * ProcessCreateExtensionStmt is executed after the extension has been
+ * created locally and before we create it on the worker nodes.
+ * As we now have access to ObjectAddress of the extension that is just
+ * created, we can mark it as distributed to make sure that its
+ * dependencies exist on all nodes.
+ */
+void
+ProcessCreateExtensionStmt(CreateExtensionStmt *createExtensionStmt, const
+						   char *queryString)
+{
+	const ObjectAddress *extensionAddress = NULL;
+
+	if (!ShouldPropagateExtensionCommand((Node *) createExtensionStmt))
+	{
+		return;
+	}
+
+	extensionAddress = GetObjectAddressFromParseTree((Node *) createExtensionStmt, false);
+
+	EnsureDependenciesExistsOnAllNodes(extensionAddress);
+
+	MarkObjectDistributed(extensionAddress);
+}
+
+
+/*
+ * PlanDropExtensionStmt is called to drop extension(s) in coordinator and
+ * in worker nodes if distributed before.
+ * We first ensure that we keep only the distributed ones before propagating
+ * the statement to worker nodes.
+ * If no extensions in the drop list are distributed, then no calls will
+ * be made to the workers.
+ */
+List *
+PlanDropExtensionStmt(DropStmt *dropStmt, const char *queryString)
+{
+	List *oldObjects = dropStmt->objects;
+
+	List *distributedExtensions = NIL;
+	List *distributedExtensionAddresses = NIL;
+
+	List *commands = NIL;
+	const char *deparsedStmt = NULL;
+
+	ListCell *addressCell = NULL;
+
+	if (!ShouldPropagateExtensionCommand((Node *) dropStmt))
+	{
+		return NIL;
+	}
+
+	/* Get distributed extensions to be dropped in worker nodes as well */
+	distributedExtensions = FilterDistributedExtensions(oldObjects);
+
+	if (list_length(distributedExtensions) <= 0)
+	{
+		/* no distributed extensions to drop */
+		return NIL;
+	}
+
+	/* Extension management can only be done via coordinator node */
+	EnsureCoordinator();
+
+	distributedExtensionAddresses = ExtensionNameListToObjectAddresses(
+		distributedExtensions);
+
+	/* Unmark each distributed extension */
```

Comment:
Super minor: In general, if the comment is a single line comment, we start with lower character. Can you go over the code to fix this minor convention issue